synapse/rest/media/v1/config_resource.py

```diff
@@ -40,9 +44,11 @@ def render_GET(self, request):
     @wrap_json_request_handler
     @defer.inlineCallbacks
     def _async_render_GET(self, request):
+        set_cors_headers(request)
         yield self.auth.get_user_by_req(request)
         respond_with_json(request, 200, self.limits_dict)
 
     def render_OPTIONS(self, request):
+        set_cors_headers(request)
         respond_with_json(request, 200, {}, send_cors=True)
```

Comment:
a) I'm surprised that we even have `OPTIONS`, since `GET` requests aren't preflighted(?), and b) I'm assuming that `send_cors=True` are actually adding the CORS headers?

Reply:
Oh yes, duh.

Reply:
![image](https://user-images.githubusercontent.com/2072976/49734805-66c38000-fc7d-11e8-976d-8076cbbb701f.png)

Riot seems to request `OPTIONS` and so I implemented it way back. I don't know if we *need* it for the request to succeed though.

Reply:
GET requests are preflighted if they have a custom Auth header, iirc